lib/maplibre-compose-material3/src/commonMain/kotlin/dev/sargunv/maplibrecompose/material3/controls/PointerPinButton.kt

```diff
@@ -0,0 +1,165 @@
+import dev.sargunv.maplibrecompose.material3.proportionalAbsoluteOffset
+import dev.sargunv.maplibrecompose.material3.proportionalPadding
+import dev.sargunv.maplibrecompose.material3.toDpOffset
+import dev.sargunv.maplibrecompose.material3.toOffset
+import kotlin.math.PI
+import kotlin.math.cos
+import kotlin.math.sin
+
+/**
+ * An elevated button in the shape of a pointer pin on the edge of an ellipsis drawn inside the
+ * parent layout, pointing towards some [targetPosition] off-screen. Only shown if the
+ * [targetPosition] is outside of the ellipsis.
+ *
+ * Note that this composable should fill the whole area in which the pointer pin button will be
+ * **placed**, i.e. usually the whole map screen.
+ *
+ * @param onClick called when this button is clicked
+ * @param targetPosition position (off-screen) the pin should point at
+ * @param modifier the [Modifier] to be applied to this button
+ * @param enabled controls the enabled state of this button. When `false`, this component will not
+ *   respond to user input, and it will appear visually disabled and disabled to accessibility
+ *   services.
+ * @param colors [ButtonColors] that will be used to resolve the colors for this button in different
+ *   states. See [ButtonDefaults.elevatedButtonColors].
+ * @param elevation [ButtonElevation] used to resolve the elevation for this button in different
+ *   states. This controls the size of the shadow below the button. Additionally, when the container
+ *   color is [ColorScheme.surface], this controls the amount of primary color applied as an
+ *   overlay. See [ButtonDefaults.elevatedButtonElevation].
+ * @param border the border to draw around the container of this button
+ * @param contentPadding the spacing values to apply internally between the container and the
+ *   content
+ * @param interactionSource an optional hoisted [MutableInteractionSource] for observing and
+ *   emitting [Interaction]s for this button. You can use this to change the button's appearance or
+ *   preview the button in different states. Note that if `null` is provided, interactions will
+ *   still happen internally.
+ */
+@Composable
+public fun PointerPinButton(
+  onClick: () -> Unit,
+  targetPosition: DpOffset,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  colors: ButtonColors = ButtonDefaults.elevatedButtonColors(),
+  elevation: ButtonElevation? = ButtonDefaults.elevatedButtonElevation(),
+  border: BorderStroke? = null,
+  contentPadding: PaddingValues = PaddingValues(12.dp),
+  interactionSource: MutableInteractionSource? = null,
+  content: @Composable (BoxScope.() -> Unit),
+) {
+  val target = targetPosition.toOffset()
+  var area by remember { mutableStateOf<Rect?>(null) }
+
+  Box(modifier = modifier.fillMaxSize().onGloballyPositioned { area = it.boundsInParent() }) {
+    val intersection = remember(target, area) { area?.let { findEllipsisIntersection(it, target) } }
```

Comment:
how hard is it to change the shape of the intersection area? probably just personal preference but I find it a bit odd the pointer floats away from the corners of the map view. Could this shape be customizable?